File: solutions/go/all-your-base/1/all_your_base.go
package allyourbase

import (
	"errors"
	"math"
	"slices"
)

func ConvertToBase(inputBase int, inputDigits []int, outputBase int) ([]int, error) {
	res := make([]int, 0)
	err := Validate(inputBase, inputDigits, outputBase)
	if err != nil {
		return res, err
	}
	pos := len(inputDigits) - 1
	dec := 0
	for _, d := range inputDigits {
		dec += d * int(math.Pow(float64(inputBase), float64(pos)))
		pos--
	}
	mod := int(math.Mod(float64(dec), float64(outputBase)))
	div := dec / outputBase
	res = append(res, mod)
	for div > 0 {
		mod = int(math.Mod(float64(div), float64(outputBase)))
		div = div / outputBase
		res = append(res, mod)
	}
	slices.Reverse(res)
	return res, nil
}

func Validate(inputBase int, inputDigits []int, outputBase int) error {
	if inputBase < 2 {
		return errors.New("input base must be >= 2")
	}
	if outputBase < 2 {
		return errors.New("output base must be >= 2")
	}
	for _, d := range inputDigits {
		if d < 0 || d >= inputBase {
			return errors.New("all digits must satisfy 0 <= d < input base")
		}
	}
	return nil
}

File: solutions/csharp/all-your-base/1/AllYourBase.cs
public static class AllYourBase
{
    public static int[] Rebase(int inputBase, int[] inputDigits, int outputBase)
    {
        if(inputBase < 2 || outputBase < 2)
            throw new ArgumentException("Input and output base must be at least 2.");
        var base10Value = ConvertFromBase(inputBase, inputDigits);
        return ConvertToBase(outputBase, base10Value);
    }

    private static int ConvertFromBase(int inputBase, int[] inputDigits)
    {
        int value = 0;
        int power = 1;
        for (int i = inputDigits.Length - 1; i >= 0; i--)
        {
            if(inputDigits[i] >= inputBase || inputDigits[i] < 0)
                throw new ArgumentException("All digits must be  andless than the input base and positives.");
            value += inputDigits[i] * power;
            power *= inputBase;
        }
        return value;
    }

    private static int[] ConvertToBase(int outputBase, int base10Value)
    {
        if (base10Value == 0) return [0];
        var digits = new List<int>();
        while (base10Value > 0)
        {
            digits.Add(base10Value % outputBase);
            base10Value /= outputBase;
        }
        digits.Reverse();
        return [.. digits];
    }
}